front/chat/src/Components/UsersList.svelte

```diff
@@ -37,23 +37,26 @@
     $: loadingSubscribersStore = mucRoom.getLoadingSubscribersStore();
 
     $: usersList = [...$usersListStore.values()] as Array<User>;
-    $: me = usersList.find((user) => user.isMe);
-    $: meArray = me ? [me] : [];
+    $: me = usersList.find((user: User) => user.isMe);
 
-    $: usersFiltered = meArray
-        .concat(
-            usersList
-                .filter((user) => user.active && !user.isMe && user.name.toLocaleLowerCase().includes(searchValue))
-                .sort((a, b) => a.name.localeCompare(b.name))
-                .concat(
-                    usersList
-                        .filter(
-                            (user) => !user.active && !user.isMe && user.name.toLocaleLowerCase().includes(searchValue)
-                        )
-                        .sort((a, b) => a.name.localeCompare(b.name))
-                )
-        )
-        .splice(0, minimizeUser ? maxUsersMinimized : usersList.length);
+    $: usersByMaps = usersList
+        .filter((user: User) => user.name.toLocaleLowerCase().includes(searchValue))
+        .reduce((reduced, user: User, index) => {
+            if ((minimizeUser && index < maxUsersMinimized) || !minimizeUser) {
+                let group = user.roomName ?? "ZZZZZZZZZZ-disconnected";
+                if (!reduced.has(group)) reduced.set(group, [user]);
+                else {
+                    const usersList = [...(reduced.get(group) ?? []), user];
+                    usersList.sort((a, b) => a.name.localeCompare(b.name));
+                    reduced.set(group, usersList);
+                }
+            }
+            return reduced;
+        }, new Map<string, User[]>());
+
+    $: roomSorted = [...usersByMaps.keys()].sort((a, b) =>
+        me?.roomName === a ? -1 : me?.roomName === b ? 1 : a.localeCompare(b)
+    );
```

Comment:
Rather than using "$:", I would tend to use derived stores.
I'm not 100% sure if this is more efficient or not to be honest so your code might be 100% ok.